File: 프로그래머스/0/120849. 모음 제거/모음 제거.swift
import Foundation

func solution(_ my_string: String) -> String {
    // 모음들을 Set으로 정의
    let vowels: Set<Character> = ["a", "e", "i", "o", "u"]

    // filter를 사용하여 모음이 아닌 문자만 남김
    let result = my_string.filter { !vowels.contains($0) }
    
    // 결과를 문자열로 변환하여 반환
    return String(result)
}

File: 프로그래머스/0/120849. 모음 제거/모음 제거.js
// function solution(my_string) {
//     var answer = my_string.split('')
//                           .filter( a => !"aeiou".includes(a))
//                           .join('');
//     return answer;
// }

// 정규표현식 사용
function solution(my_string) {
    return my_string.replace(/[aeiou]/g, '')
}

// // 'aeiou'가 연속적으로 나오는 부분만 제거하고 싶다면 배열 제거
// // 정규표현식은 문자열에서 특정 패턴을 찾고, 수정하는 도구이기 때문에 // 사이에 ""따옴표 필요 없음
// function solution(my_string) {
//     return my_string.replace(/aeiou/g, '');
// }
